crates/rooch-framework/sources/account_authentication.move

```diff
@@ -37,6 +37,12 @@ module rooch_framework::account_authentication{
       }
    }
 
+   #[private_generics(ValidatorType)]
```

Comment:
Entry function can not limit the `ValidatorType` via private_generics

Reply:
The entry function of the rotate_authentication_key should be in the validator module; the validator maybe needs to do some checks before updating the authentication key.

Reply:
If we want to provide a rotate_authentication_key_entry function to end-users, we can provide a private function `rotate_authentication_key_internal`, and no `private_generics` limit. the `rotate_authentication_key` and `rotate_authentication_key_entry` call the rotate_authentication_key_internal function.

Reply:
Understand.

Reply:
> If we want to provide a rotate_authentication_key_entry function to end-users, we can provide a private function `rotate_authentication_key_internal`, and no `private_generics` limit. the `rotate_authentication_key` and `rotate_authentication_key_entry` call the rotate_authentication_key_internal function.

Because it will be a warning if calling it a normal function `rotate_authentication_key`:
```zsh
2023-08-03T22:50:32.253423Z  WARN connection{remote_addr=127.0.0.1:57018 conn_id=1}: moveos::moveos: execute tx(0x2ad8…406c) error: Err(VMError { major_status: EXECUTE_ENTRY_FUNCTION_CALLED_ON_NON_ENTRY_FUNCTION, sub_status: None, message: None, exec_state: None, location: Module(ModuleId { address: 0000000000000000000000000000000000000000000000000000000000000003, name: Identifier("account_authentication") }), indices: [(FunctionDefinition, 4)], offsets: [] })    
```

Reply:
I've done these:

1. Removed ValidatorType for entry function.
2. Added checks and moved rotate_authentication_key_entry to validator modules.
3. `rotate_authentication_key_internal` is a public(friend) function in account_authentication.move, and I could not call it from other crates since it has restrictions. So I made `rotate_authentication_key_entry` to call `rotate_authentication_key` public function.